app/assets/javascripts/clipboard.js

```diff
@@ -0,0 +1,37 @@
+$(document).on("turbolinks:load", function () {
+  addCopyButton();
+});
+
+function addCopyButton() {
+  new Clipboard('.clipboard-copy');
+
+  $("pre").each(function(index) {
+    var id = "id_" + index;
+    $(this).children("code").attr("id", id);
+    prependButtonAndHide($(this), id);
+    toggleCopyButton($(this));
+  });
+}
+
+function prependButtonAndHide(pre, id) {
```

Comment:
it's convention to name variables that are jquery objects with a dollar sign in front of them `$pre` but not necessary.